### PR TITLE
Clean up some linting and style

### PR DESCRIPTION
**Pull request by elfiesmelfie**

### CODE REVIEW

File: roles/client_side_tests/tasks/test_e2e.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+- name: "Extract URL from stf-connectors template"
+  shell: grep interconnect /home/stack/virt/stf-connectors* | sed 's;.*\(-service-telemetry.*\);default-prometheus-proxy\1;'
+  register: prom_url
+  changed_when: false
+  failed_when: "prom_url.stdout_lines|length == 0"
+
+- name: "[tmp] Set prom creds"
+  set_fact:
+    prom_user: "internal"
+    prom_pass: "1tp9zXur-t8Ve,wPkVgE"
```
[the remark this answers is not in the document]
Comment (elfiesmelfie) [reply roles/client_side_tests/tasks/test_e2e.yml:11]: That's good to know, I was wondering where it came from!

[the remark this answers is not in the document]
Comment (elfiesmelfie) [reply roles/client_side_tests/tasks/test_e2e.yml:11]: I'm going to move the prom_user and prom_secret to the defaults/main.yml for now, and update it in the future so you need to pass them in.

I'll add documentation to explain how to get that info.  I think we could add some set-up tasks to the role as well, for info gathering. For now, I'll add the ability to pass the values of prom_user and prom_pass, and use the stf_functional_test.yml to gather that info. Eventually, I want to minimise the amount of info that needs to be passed, and just let the user pass a set of oc credentials (e.g. username and password), so we can gather the other info automatically.

File: roles/client_side_tests/tasks/test_e2e.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
+---
+- name: "Extract URL from stf-connectors template"
+  shell: grep interconnect /home/stack/virt/stf-connectors* | sed 's;.*\(-service-telemetry.*\);default-prometheus-proxy\1;'
```
Comment (elfiesmelfie) [inline roles/client_side_tests/tasks/test_e2e.yml:3]: @lnatapov, does this work to get the same info? 
    oc get route/default-prometheus-proxy -ojson | jq '.spec.host'

If so, I can update this to allow passing in the var, and add a when: to this task to run only if it isn't already defined.

Comment (lnatapov) [reply roles/client_side_tests/tasks/test_e2e.yml:3]: @elfiesmelfie yes. if you run the following command it will give you the same result but it will give you the promethues URL with double quotes.

Example:
[quicklab@upi-0 ~]$ oc get route/default-prometheus-proxy -ojson | jq '.spec.host'
"default-prometheus-proxy-service-telemetry.apps.leonidcluster.lab.upshift.rdu2.redhat.com"

In order to remove the quotes and get clean URL run this:

[quicklab@upi-0 ~]$ oc get route/default-prometheus-proxy -ojson | jq '.spec.host' |  sed 's/^"\|"$//g'
default-prometheus-proxy-service-telemetry.apps.leonidcluster.lab.upshift.rdu2.redhat.com

File: roles/test_alerts/tasks/first_test.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            rules:
+              - alert: Collectd metrics receive rate is zero
+                expr: rate(sg_total_collectd_msg_received_count[1m]) == 0
+      EOF
+
+- name: "Check that the alert was created"
+  shell:
+    cmd: |
+      curl -k --user "{{ prom_user }}:{{ prom_pass }}" https://{{ prom_host }}/api/v1/rules
+  register: output
```
Comment (csibbitt) [inline roles/test_alerts/tasks/first_test.yml:31]: WIP? I guess I don't see this called from anywhere yet either

**Review by csibbitt — COMMENTED** [review]
Just reviewing the progress, everything LGTM so far

File: roles/test_alerts/tasks/first_test.yml (hunk 3 above)
Comment (elfiesmelfie) [reply roles/test_alerts/tasks/first_test.yml:31]: This is what I demoed on Monday. I am testing with an existing environment, and figuring out what needs to go where for set-up i.e. What is generic enough to go here, and what is something that will be passed as vars  by the calling playbook.
I plan to add some playbooks in the collection as a reference to run each role.

File: galaxy.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+namespace: 'infrawatch'
+name: 'functional_tests'
```
[the remark this answers is not in the document]
Comment (elfiesmelfie) [reply galaxy.yml:3]: I am open to new suggestions, since since these are end-to-end tests, as well as some component functional tests too.
If there's a good suggestion, I would recommend that we also change the name of this repo, to avoid confusion with naming.

[the remark this answers is not in the document]
Comment (lnatapov) [reply galaxy.yml:3]: STF test suite ?
We can split tests into 3 categories. OSP tests is when we test that collectd and metrics_qdr containers are up and running on the OSP overcloud nodes and have metrics locally, then we have STF tests. These are done when we run smoke tests and finally e2e or integration tests when both OSP and STF are involved in the test.  There also will be some STF related tests that will be out of smoke tests scope e.g. prometheus rule creation test that it's purely STF test but not part of smoke tests

[the remark this answers is not in the document]
Comment (elfiesmelfie) [reply galaxy.yml:3]: So, client-side, server-side and end-to-end?

I'd also be inclined to split the tests by scope: single component (incl. regression tests where we check for specific bugs), functional tests for a component, similar to smoketests, but with the actual services generating the data and end to end tests.

I don't think the two lists are mutually exclusive, and we'll need some more granular tests as well.
We can define really granular tests, with a tight scope per role, and then use the playbooks to define test suites 
For example (some hypothetical tests here)
* test-osp-components playbook could include the collectd and qdr roles;
* test-stf would include testing-alerts and testing that the queries we have defined for the dashboards return data; * test end-to-end may include various scenarios including the prom tests we have, but also an autoscaling scenario, if we decide to add that).

TL;DR: we can use roles to define the individual test, and use playbooks to group them.

**Review by lnatapov — COMMENTED** [review]
Hey @elfiesmelfie .
I have created a role and test for metrics retention. Can you please review this commit

**Comment by elfiesmelfie** [comment]
> Hey @elfiesmelfie . I have created a role and test for metrics retention. Can you please review this commit

As discussed, I have reset this branch back to before your changes.
I will merge this branch, since we're satisfied with the state it was then.
Once this PR is closed, you can re-upload your changes and open a PR against master
